array.c

```diff
@@ -5101,6 +5075,71 @@ rb_ary_compact(VALUE ary)
     return ary;
 }
 
+/*
+ *  call-seq:
+ *     ary.intersection(other_ary1, other_ary2, ...)      -> new_ary
+ *
+ *  Set Intersection --- Returns a new array containing unique elements common
+ *  to +self+ and <code>other_ary</code>s. Order is preserved from the original
+ *  array.
+ *
+ *  It compares elements using their #hash and #eql? methods for efficiency.
+ *
+ *     [ 1, 1, 3, 5 ].intersection([ 3, 2, 1 ])                    # => [ 1, 3 ]
+ *     [ "a", "b", "z" ].intersection([ "a", "b", "c" ], [ "b" ])  # => [ "b" ]
+ *     [ "a" ].intersection #=> [ "a" ]
+ *
+ *  See also Array#&.
+ */
+
+static VALUE
+rb_ary_intersection_multi(int argc, VALUE *argv, VALUE ary)
+{
+    VALUE hash, v, result, current, shortest;
+    st_data_t vv;
+    long i, j, min;
+
+    shortest = ary;
+    min = RARRAY_LEN(ary);
+    for (i = 0; i < argc; i++) {
+        current = to_ary(argv[i]);
```

Comment:
@nobu perhaps you could advise me. I'd like to sort `argv` by the size of the input arrays, but I'm not sure how to accomplish that with something like `rb_ary_sort` which seems to sort them recursively.

Reply:
```c
static int
sort_by_size(const void *a, const void *b, void *dummy)
{
    long alen = RARRAY_LEN(*(const VALUE *)a);
    long blen = RARRAY_LEN(*(const VALUE *)b);
    if (alen < blen) return +1;
    if (alen > blen) return -1;
    return 0;
}

{
    // ...
    for (i = 0; i < argc; i++) argv[i] = to_ary(argv[i]);
    ruby_qsort(argv, argc, sizeof(VALUE), sort_by_size, NULL);
    // ...
}
```

Reply:
Thank you!